pkg/router/routeapihelpers/validation.go

```diff
@@ -122,6 +122,33 @@ func sanitizePEM(data []byte) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// splitCertKey takes a slice of bytes containing sanitized PEM data and returns
+// two slices of bytes containing PEM data: one slice with the public
+// certificate block or blocks from the input PEM data and one slice with any
+// private key blocks.
+func splitCertKey(data []byte) ([]byte, []byte, error) {
+	var block *pem.Block
+	publicBuf := &bytes.Buffer{}
+	privateBuf := &bytes.Buffer{}
+	for len(data) > 0 {
+		block, data = pem.Decode(data)
+		if block == nil {
+			break
+		}
+		switch block.Type {
+		case "PUBLIC KEY", "CERTIFICATE":
+			if err := pem.Encode(publicBuf, block); err != nil {
+				return nil, nil, err
+			}
+		case "EC PRIVATE KEY", "RSA PRIVATE KEY":
```

Comment:
(im probably just out of the loop here, but...) Are EC and RSA the only acceptable types of private keys here?

Reply:
That's right.  More to the point, `splitCertKey` expects *sanitized* PEM data, meaning PEM data that came from `sanitizePEM`, and `sanitizePEM` (by way of `knownBlockDecoders` and `privateKeyBlockVerifier`) always uses either the block type `"RSA PRIVATE KEY"` or `"EC PRIVATE KEY"` for private keys.